content/well-architected-framework/docs/docs/secure-systems/identity-access-management/create-permissions-guardrails.mdx

```diff
@@ -0,0 +1,67 @@
+---
+page_title: Create permissions and guardrails
+description: Write effective identity and access management (IAM) policies that enforce permissions guardrails by using least privilege.
+---
+
+# Create permissions and guardrails
+
+Now that you have defined your access requirements, you can begin writing
+policies that enforce permissions guardrails by using least privilege policies.
+
+Creating policies and enforcing guardrails is the process of translating
+requirements into policies that define what users, applications, and systems can
+do. This process involves converting the requirements for each
+role or function within your organization when you [defined your access
+requirements](/well-architected-framework/secure-systems/identity-access-management/define-access-requirements)
+into policies for your systems.
+
+## What are permissions
+
+Permissions are the specific actions that users, applications, or systems can
+take on resources. Permissions are typically defined in policies and used to
+control access to resources in a system.
+
+You can grant or deny permissions based on a variety of factors, including
+the user's role, group membership, the resource they are attempting to access,
+and the context of the access request. For example, a user may have permission
+to read data from a database, but not to write data to the database.
+
+In practice, this policy for a database might look like:
+
+<CodeBlockConfig hideClipboard>
+
+```plaintext
+GRANT SELECT ON ALL TABLES IN SCHEMA public TO readonly_user;
+```
+
+</CodeBlockConfig>
+
+The PostgreSQL role `readonly_user` with this permission can select data from
+tables, but not write or modify data.
+
+## Define permissions as code
+
+Where possible, define your permissions in policies as code. Defining
+permissions in policies as code allows you to version control your policies,
+track changes over time, and collaborate with your team. You can then use tools like
+HashiCorp's Terraform or Red Hat's Ansible to deploy and update your policies.
+Most systems support writing policies in JSON, YAML, or XML.
+
+Some systems like HashiCorp's Vault and Sentinel support policies in custom languages like
+the HashiCorp Configuration Language (HCL). JSON is another widely supported
+format. HashiCorp Vault supports JSON based policies, in addition to HCL. AWS
+Identity and Access Management (IAM) supports JSON-based policies. Other systems
+like Active Directory may require scripting the policies in other languages like
+PowerShell.
+
```

Comment:
I would add a code example to show users how a simple permission as code would look.